### ISSUE TITLE
PWA ERR_FAILED on refresh

### ISSUE THREAD

**Issue by PeterMarques** [godotengine/godot#69414]
### Godot version

3.5.1 stable

### System information

Androids and W10

### Issue description

I think that the culprit may be on the Cloudflare Pages, but i may only be pointing finger cus the "Workers" tag is a common theme.

Well, when exporting HTML5 with PWA enabled, i can install it on Android phones, and W10 machines.

But, when trying to acess the app, for the second, something third or forth time, i get the unescapable error, that only recovers after i purge the cache of the app.
![image](https://user-images.githubusercontent.com/32851924/204909205-5bf659db-8ec7-4360-82b6-870f998820e2.png)

Thats on windows, but on android phones it happens the same error, and when tring to load from browser it happens as well sometimes.(if the PWA is already installed)

There are several factors:
Overall, this is a know bug, from 2019, but it seens to be fixed in the atual webview implementation of chromium.

There is the fact that the PWA architecture of godot was recently modified to adequate to some more networky sane behaviour.

There is even the Cloudflare Pages, that i am using to host the export, that, -Maybe- is causing the conflict.

Thanks for any insights.

### Steps to reproduce

I think this is reproductible in a whole manner if someone using G3.5.1S exports a HTML5 page with PWA and host it on Cloudflare Pages, and set a custom domain. Thas my setup that is generating the error.

### Minimal reproduction project

N/A

**Comment by PeterMarques** [comment]
On the W10 i could press f12 and get more insightful data.
The FetchEvent for "(the link)" resulted in a network error response: a redirected response was used for a request whose redirect mode is not "follow".

Again, that may indicate something on the cloudflare dns servers then, maybe because of the custom domain? I will try to install using the deploy domain, just in case .

**Comment by PeterMarques** [comment]
On the W10 i could press f12 and get more insightful data.

"
The FetchEvent for "(the link)" resulted in a network error response: a redirected response was used for a request whose redirect mode is not "follow".
"
and
"
The service worker navigation preload request was cancelled before 'preloadResponse' settled. If you intend to use 'preloadResponse', use waitUntil() or respondWith() to wait for the promise to settle.
"
the above, several times

and
"
crbug/1173575, non-JS module files deprecated.
(anonymous) @ (index):6789
"
Again, that may indicate something on the cloudflare dns servers then, maybe because of the custom domain? But installing using the deploy domain results in the same behaviour.

Nothing, after second or third openning, same error.

**Comment by Calinou** [comment]
> There is the fact that the PWA architecture of godot was recently modified to adequate to some more networky sane behaviour.

Can you reproduce this on older Godot versions such as 3.3.4 and 3.4.4?

**Comment by PeterMarques** [comment]
3.3.4 does not seens to have PWA yet.

I will only do the 3.4.5 then, you sugest another versions?

Will try to host on another service. But it needs to be free, no trial etc, any sugestions?

**Comment by Calinou** [comment]
> Will try to host on another service. But it needs to be free, no trial etc, any sugestions?

GitHub Pages or itch.io.

**Comment by PeterMarques** [comment]
I will try that, but it seen to be in need of a regression fix, not a hosting issue.

I created a null project, just to try the pwa effect

It was reproductibe on the 3.5.1 of course, but on the 3.4.5 it was not. It is working on the older version

Looking in the JS file of the worker, we see several changes.

https://godotpwatest.pages.dev/ - 3.4.5 - working

https://godotpwatestbug.pages.dev/ - 3.5.1 - BUG!

I tried using the https://text-compare.com/ to make the diff

But, i cant post the .JS file here, not acepted.

Whats the best way to send both 3.4.5 and 3.5.1 files here for comparison?

And, can you comprove the bug link is indeed not working after installing and opening the instaled version several times?

Thanks for the support.

**Comment by PeterMarques** [comment]
PS. dont try to install both version at the same time. Its the same name same project, it seens to not install.. Instal one, then unnistall then install another. Sorry. (and that is another bug, diferent URL with the same project name is getting ignored on installation)

**Comment by PeterMarques** [comment]
I rewrited this several times, because several times i believed i had fixed it.

In a last atempt, i simple attended to the error that said i need to have a "follow".

And i changed:

const isNavigate = event.request.mode === "navigate";
to
const isNavigate = event.request.mode === "follow";

and

if (event.request.mode === "navigate") {
to
if (event.request.mode === "follow") {

And the worker worked again.

Links:
https://godotpwanotfix.pages.dev/ - 3.5.1 export
https://godotpwanotfix.pages.dev/index.service.worker.js - 3.5.1 worker direct link for ease of comparison

https://godotpwafixfinal.pages.dev/ - 3.5.1 export modified worker
https://godotpwafixfinal.pages.dev/index.service.worker.js - 3.5.1 modified worker direct link for ease of comparison


If that creates security concerns, or breaks another thing, i do not know.

But, aparently it is working on windows.

Will try custom domains and on android phone, to se if all is good now.

Is that a aceptable solution? Because on the 3.4 version that worked, it was not using the "follow" atribute.

**Comment by PeterMarques** [comment]
It indeed seens to be working on custom domain and android.

So, i just need a confirmation if this is indeed a bug, or just a odd behaviour, and if the follow instead of navigate is the right solution.

Thanks.

**Comment by PeterMarques** [comment]
Soo. Another thing. 
None Works offline.
Not the 3.4.5 not the 3.5.1 not the fixed 3.5.1.
And everyone is a diferent error

3.4.5 link/ missing
3.5.1 link/Index.html missing
3.5.1 fixed Android error That i need do be online once for the First time and then sometimes a Brower text simple HTML saing that the browser is missing some functionality.

**Comment by PeterMarques** [comment]
And another thing, how is the new-version trigger? Because with the new model of download, the first time you open it, it still displays the older version, and only after a second opening the new version displays.

There is a way to make it so that if there is a new version, it load that first time, even if it takes longer.? Having a old version available a single time can be a door to abuse, if a bugfix was released, or anything like that.

**Comment by PeterMarques** [comment]
So, working again on PWA, new app, on latest godot3, same behaviour. I just googled the error and it poped up my own issue, so...

No one regularly enconters that behaviour? It instaled fine, opened just one time perfeclt, then, only error.

Is that a very particular edge case of cloudflare pages?

**Comment by PeterMarques** [comment]
and it inded seen to be a fix as described, to change the "navigate" instances to "follow", it continue to works, after several opens.

i do not know how it will work on changes, but will verify, but it indeed made the verify pwa func useless because it seens to in this mode to aways re-cache the pages, and again as previously describided, it does not works if the phone has no internet acess.

So, it renders the pwa useless and it just became a full-screen shortcut, but thats the state.

how to improve that?

**Comment by PeterMarques** [comment]
Now, more weird behaviours;

as stated, i use cloudflare to upload.

I started to try on several modification of the file, and eventually i get myself on the ERR_FAILED again, something that was "fixed".

Ok, lets just change my file again, delete cache, reinstal pwa, and it should work.

NOPE

Ok, lets ROLLBACK the pages to a instance that is garantee to work, and even compare source files

NOPE, again i am on the ERR_FAILED, and now it wont load and stays on the ERR_FAILED.

A purged cache/data reinstall, that goes back to a state of garantee functionality, and still errror?

This worker is really a PITA.

I suspect https://github.com/godotengine/godot/issues/75743#issuecomment-1553036695 may be a workarround, but how to do that on a PWA mobile/android?

**Comment by PeterMarques** [comment]
Closing since 3.x on web is no longer relevant on my usage.
